accelerator/core/core.c

```diff
@@ -581,3 +583,33 @@ status_t ta_get_iri_status(const iota_client_service_t* const service) {
 
   return ret;
 }
+
+status_t ta_update_iri_conneciton(ta_config_t* const ta_conf, iota_client_service_t* const service) {
+  status_t ret = SC_OK;
+  for (int i = 0; i < MAX_IRI_LIST_ELEMENTS && ta_conf->host_list[i]; i++) {
+    // update new IRI host
+    ta_conf->host = ta_conf->host_list[i];
+    ta_conf->port = ta_conf->port_list[i];
+    service->http.host = ta_conf->host_list[i];
+    service->http.port = ta_conf->port_list[i];
+    ta_log_info("Try tp connect to %s:%d\n", ta_conf->host, ta_conf->port);
+
+    if (iota_client_core_init(service)) {
+      ta_log_error("Initializing IRI connection failed!\n");
+      return SC_TA_OOM;
```

Comment:
Amend the error code.
When iota_client_core_init failed, it will return RC_CCLIENT_UNIMPLEMENTED.

Reply:
You mean append it in the error message in `line 598`?

Reply:
I think it is not `SC_TA_OOM`.